stock_move_auto_assign_auto_release/models/stock_move.py

```diff
@@ -57,19 +57,38 @@ def _search_is_auto_release_allowed(self, operator, value):
             domain = [("id", "not in", self.search(domain).ids)]
         return domain
 
-    def _enqueue_auto_assign(self, product, locations, **job_options):
-        job = super()._enqueue_auto_assign(product, locations, **job_options)
+    def _filter_auto_releaseable_locations(self, locations):
+        all_warehouses = self.env["stock.warehouse"].search([])
+        return self.env["stock.location"].search(
+            [
+                ("location_id", "child_of", all_warehouses.view_location_id.ids),
+                ("id", "in", locations.ids),
+            ]
+        )
```

Comment:
this is a little bit less costly I think (to reduce the amount of search as we are in a for loop)
```suggestion
        wh_parent_paths = all_warehouses.view_location_id.mapped("parent_path")
        return locations.filtered(lambda loc: any(loc.parent_path.startswith(wh_parent_path) for wh_parent_path in wh_parent_paths))
```